### DISCUSSION TITLE
How to change name of our app

### DISCUSSION THREAD

**Discussion by tyroneBaba** [appcelerator/titanium_mobile#12955] [Q&A]
Hello Ti devs,

We need to change the name of our app. Can somebody confirm if changing the name property in tiapp.xml is what we want and also to change the name in the play store/app store or if there is more required?

https://titaniumsdk.com/api/titanium/app.html#name

[Old stackoverflow question](https://stackoverflow.com/questions/7333422/how-to-change-the-name-of-an-app), not sure how accurate this is as its old but is this still relevant?

Thanks

**Comment by m1ga** [comment]
the `name` (or `app/i18n/en/app.xml` if you are using i18n names) is responsible for the name of the installed app. Inside the stores you'll have to do the other changes. You'll find those places inside the store help pages (not related to Titanium so you'll find a lot of help there)

**Reply by tyroneBaba** [reply]
I see ok, on iOS we're having a slight problem with having a space between words, do you know how to go about doing this? people say to add an escape character for it.. this is only a minor issue though.

On android, it has created 2 apps with the new name and one app doesn't open at all and gets stuck on the splash screen. Do you have any idea as to whats happening here?

**Reply by m1ga** [reply]
https://devblog.axway.com/mobile-apps/internationalization-of-app-names/
thats what the i18n file is for too

don't change the bundleid that would produce a second app

**Reply by gjerlow** [reply]
Also make sure that the `android:name` property of the `activity` element in your tiapp is correct after you change the app name. That could cause a double app icon where one of them is just an "empty app". (Might not be needed if you use i18n?)

**Reply by tyroneBaba** [reply]
@gjerlow do you have an example of this? trying to find out where but my build keeps failing so i don't think im doing it in the right spot.

**Reply by m1ga** [reply]
if it is not in your tiapp.xml then you probably didn't set it there before so you don't need to do it there.
What do you mean by build is failing? Is there an error? If you change the name you'll have to do a `ti clean` so you don't have duplicate xcode projects for iOS.

**Reply by tyroneBaba** [reply]
```
[INFO]  [GRADLE] > Task :app:desugarDebugFileDependencies
[ERROR] [GRADLE] 
[ERROR] [GRADLE] FAILURE: Build failed with an exception.
[ERROR] [GRADLE] 
[ERROR] [GRADLE] * What went wrong:
[ERROR] [GRADLE] Execution failed for task ':app:processDebugResources'.
[ERROR] [GRADLE] > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
[ERROR] [GRADLE]    > Android resource linking failed
[ERROR] [GRADLE]      /Users/(appname)/build/android/app/build/intermediates/packaged_manifests/debug/AndroidManifest.xml:51: AAPT: error: attribute 'android:name' in <activity> tag must be a valid Java class name.
[ERROR] [GRADLE]          
[ERROR] [GRADLE] 
[ERROR] [GRADLE] * Try:
[ERROR] [GRADLE] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[ERROR] [GRADLE] 
[ERROR] [GRADLE] * Get more help at https://help.gradle.org
[ERROR] [GRADLE] 
[ERROR] [GRADLE] BUILD FAILED in 8s
[INFO]  [GRADLE] 93 actionable tasks: 93 executed
[ERROR] "gradlew" tool returned exit code: 1
Process exited with 1
The terminal process terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

**Reply by m1ga** [reply]
what is the name in that file `/Users/(appname)/build/android/app/build/intermediates/packaged_manifests/debug/AndroidManifest.xml` line 51?

**Reply by tyroneBaba** [reply]
"org.appcelerator.titanium.TiActivity"

yeah, probably in the wrong place, eh? haha

**Reply by tyroneBaba** [reply]
Ok, so we've tracked our problem down to our Activities in our Tiapp.xml.. we have an activity but the autogenerated AndroidManifest.xml is showing that we have 2 activities of the same kind which is why its making two apps in the app drawer.  We remove our custom activity from tiapp.xml but now we don't have the custom properties we set. is there some way to set these properties? we don't seem to have any control over the autogenerated activity.

**Reply by m1ga** [reply]
it's very difficult if you do custom stuff :smile: How did you create the <activity> node in the first place? Just do it now again with the new name. Normally you'll do https://wiki.appcelerator.org/display/guides2/Android+Themes#AndroidThemes-Overrideanactivitytheme
The name should be visible in the activity name. But again: don't know what you are doing and why you have custom things.

**Reply by jquick-axway** [reply]
@tyroneBaba , I'm not sure what you're doing. Are you trying to define an activity like this?
```xml
<ti:app>
	<android>
		<activities>
			<activity android:exported="true" url="MyActivity.js"/>
		</activities>
	</android>
</ti:app>
```

Defining an activity like the above is a dead concept as of Titanium 8.0.0. While the build system will create a Java activity derived class for the above and define it in the generate AndroidManifest.xml" file, Titanium 8.0.0 will no longer allow that custom activity to be displayed. Titanium will destroy and it instead redirect the end-user to the main `TiRootActivity` (the splash activity) UI stack to make it work more like iOS (kind of like how Google's "singleTask" feature works).

You "can" customize an "existing" activity in the manifest section of the "tiapp.xml" such as `TiActivity`, a 3rd party library's activity, etc. This still works. It correctly merges a chain of AndroidManifest.xml files between the library, app, and custom manifest file as defined [here](https://developer.android.com/studio/build/manifest-merge#merge_priorities).
...What won't work is customizing the activity you defined as shown above in the `<manifest/>` section. That will cause the duplication you ran into. But there's really no point in doing that anymore anyways.
